docker/Dockerfile

```diff
@@ -1,33 +1,42 @@
-FROM python:3.9.16-slim
-
-# Environment Variables
-ENV ARGILLA_HOME_PATH=/var/lib/argilla
-ENV DEFAULT_USER_ENABLED=true
-ENV DEFAULT_USER_PASSWORD=1234
-ENV DEFAULT_USER_API_KEY=argilla.apikey
-ENV USERS_DB=/config/.users.yml
-ENV UVICORN_PORT=6900
-
-# Copying script for starting argilla server
-COPY scripts/start_argilla_server.sh /
+FROM python:3.9-slim AS builder
 
 # Copying argilla distribution files
 COPY dist/*.whl /packages/
-
+RUN python -m venv /my-virtual-env
+ENV PATH="/my-virtual-env/bin:$PATH"
 RUN apt-get update && \
-    apt-get install -y python-dev libpq-dev gcc && \
-    chmod +x /start_argilla_server.sh && \
+    apt-get install -y python-dev-is-python3 libpq-dev gcc && \
     for wheel in /packages/*.whl; do pip install "$wheel"[server,postgresql]; done && \
-    apt-get remove -y python-dev libpq-dev gcc && \
+    apt-get remove -y python-dev-is-python3 libpq-dev gcc && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/* && \
     rm -rf /packages
 
+
+FROM python:3.9-slim
+
+# Environment Variables
+ENV ARGILLA_HOME_PATH=/home/worker/argilla
+ENV DEFAULT_USER_ENABLED=true
+ENV DEFAULT_USER_PASSWORD=1234
+ENV DEFAULT_USER_API_KEY=argilla.apikey
+ENV USERS_DB=/config/.users.yml
+ENV UVICORN_PORT=6900
+
+RUN useradd -ms /bin/bash worker
+COPY scripts/start_argilla_server.sh /home/worker
+COPY --chown=worker:worker --from=builder /my-virtual-env /home/worker/my-virtual-env
+ENV PATH="/home/worker/my-virtual-env/bin:$PATH"
+
 # Create argilla volume
 RUN mkdir -p "$ARGILLA_HOME_PATH"
```

Comment:
I've been doing some tests but I found problems applying the database migration. Argilla's home folder is created by `root` user. So, when Argilla is running, the `worker` user has no permissions to execute database migrations (and all write-db-related actions) in `/home/worker/argilla`

Reply:
After fixing permission problems, the db migration task can be applied, but I found an error pointing to a missing file:

```bash
docker-argilla-1        | start_argilla_server.sh: line 19: /home/worker/my-virtual-env/bin/uvicorn: cannot execute: required file not found
```

Reply:
Hey! Thanks for doing all that and explaining the sqlite thing.

I'm not sure why that has happened tbh, I did notice that before. I got round it when testing by just doing `python -m uvicorn` rather than `uvicorn`. 

Probably need to do something clever when copying the virtual environment over in the multistage build or it might be simpler to just install uvicorn as part of the second stage docker build so it's on path.

Please feel free to push commits/makes changes to my code if that's faster/easier. Otherwise will take a look when I next get time, hopefully tomorrow or next week.